### ISSUE TITLE
Curser Jumping when entering quotation marks using markdown

### ISSUE THREAD

**Issue by Basti-Fantasti** [laurent22/joplin#6256]
## Environment

Joplin version: 2.7.13 prod,win32
Platform: Windows
OS specifics: Windows 10, German Language, German Keyboard

## Steps to reproduce

The thread leading to this bug report can be found in the [joplin forum](https://discourse.joplinapp.org/t/issue-when-entering-quotation-marks-with-markdown/24312)

@JackGruber also added a [short video](https://discourse.joplinapp.org/t/issue-when-entering-quotation-marks-with-markdown/24312/6) which shows the issue 

1.  Start a note with a headline
2.  go to next line and try to enter the quotation mark `for this highlighting`
3.  just start typing directly with the quotation mark followed by e.g. the word **for**
4. then the f will be added after the quotation mark and the cursor jumps back to the beginning of the note

## Describe what you expected to happen

That I can use regular markdown syntax with typical writing 😄 

## Logfile

None available, but the video as linked above!

**Comment by laurent22** [comment]
I can't replicate this. Please could you provide a video showing the issue?

**Comment by JackGruber** [comment]
I try to Write

    Hello `world`

![Joplin_CfWWnMyYCV](https://user-images.githubusercontent.com/24863925/157862001-5cdc6644-69d2-46ce-906f-b89e396d67c6.gif)

No plugins activated. 

Windows 11
```
Joplin 2.7.14 (dev, win32)

Client ID: 0a26e311ccea4bd5903ab8e3fd82597f
Sync Version: 3
Profile Version: 41
Keychain Supported: Yes

Revision: 75cc9d7
```

This habbens also with the PortableVersions (Tested down to version `2.4.9`), but I never noticed the error before.

**Comment by laurent22** [comment]
Hmm, I still can't replicate it on macOS, but backticks are special on French keyboards. Which layout are you using?

**Comment by JackGruber** [comment]
A German QWERTZ Layout.

**Comment by Daeraxa** [comment]
Tried to replicate on Linux but don't seem to be able to. Enabled German keyboard layout and tried to follow the steps (with the assumption that the backtick key is the one next to backspace and acts as a dead key). Pressed shift + \` twice to get it to print but it worked correctly from what I can see.

**Comment by roman-r-m** [comment]
Can't replicate either (Linux, no German layout)

**Comment by laurent22** [comment]
Ok I can replicate it with QWERTZ layout on Windows. The sequence of key is `h, e, l, l, o, Space, Shift+Backtick, w`. So the W is pressed while the backtick hasn't been printed yet. I wonder if it's some CodeMirror shortcut that's being triggered?

**Comment by Daeraxa** [comment]
That gets me something very different on Linux - `hello ẁ`

**Comment by laurent22** [comment]
Actually it does it with the French layout too - `Altr Gr + 7, W`. I'm used to always press space after backtick, so that it prints it, but if you press the next letter directly, it goes back at the beginning.

**Comment by JackGruber** [comment]
Thats True, after pressing a space the backtick is shown and the cursor dosn't jump.

**Comment by Daeraxa** [comment]
So a space works fine too, basically whatever it is doesn't seem to be an issue on Linux (or at the very least Linux Mint Cinnamon).
`h, e, l, l, o, Space, Shift+Backtick, Space, w` and `h, e, l, l, o, Space, Shift+Backtick, Shift+Backspace, w` both produce a normal output -  hello `world`

@JackGruber out of interest what was your exact keypress combo?

**Comment by JackGruber** [comment]
Cursor jumps: `h` `e` `l` `l` `o` `SPACE` `Shift + Backtick` `w`
No curser jump: `h` `e` `l` `l` `o` `SPACE` `Shift + Backtick` `SPACE` `w`

**Comment by laurent22** [comment]
This seems to be Windows only because I cannot replicate it on macOS either.

**Comment by shbach** [comment]
I had a similar bug when using [Espanso](https://github.com/federico-terzi/espanso) but it's not triggered by backtick marks. I [reported it as an Espanso bug](https://github.com/federico-terzi/espanso/issues/1013) but perhaps it's related to this.

**Comment by JakubGIT** [comment]
I've been experiencing the same issue.
I haven't found this one, so I started #6275 

a few remarks:
1. safe mode solves the problem
2. german keyboard layout as well as the czech one suffers from the problem
3. whereas german and czech keyboards insert one backtick, when backtick pressed (<kbd>shift</kbd> + <kbd>´</kbd> has to be pressed actually), english layout works differently: when I press backtick: <kbd>`</kbd>, two backticks are written, just like by the "code" keyboard shortcut (default <kbd>ctrl</kbd> + <kbd>alt</kbd> + <kbd>c</kbd> I believe)

**Comment by Daeraxa** [comment]
> when I press backtick: `, two backticks are written, just like by the "code" keyboard shortcut (default ctrl + alt + c I believe)

Is that not just because of the option in config > note `Auto-pair braces...etc.`
Curious that the same option being enabled has different behaviours just because of how the character was inserted on the keyboard.

**Comment by JakubGIT** [comment]
> > when I press backtick: `, two backticks are written, just like by the "code" keyboard shortcut (default ctrl + alt + c I believe)
> 
> Is that not just because of the option in config > note `Auto-pair braces...etc.` Curious that the same option being enabled has different behaviours just because of how the character was inserted on the keyboard.

I made a mistake, it's actually like that:
#### keyboard layouts
whereas german and czech keyboards insert one backtick, when backtick pressed (shift + ´ has to be pressed actually) and then another key is pressed (can be a letter or a space or another backtick, english layout works differently: when I press backtick: `, two backticks are immediately inserted, just like by the "code" keyboard shortcut (default ctrl + alt + c I believe)

#### other applications
but - other applications react like that too - I mean - in english a backtick is immediatelly inserted, in german and czech a backtick waits for another key. It's actually pretty logical, because it's primarily *not a backtick*, but a reverse axon. Like è à ì, etc. (although so far I know, reverse axon is not used in any of those two languages, but a normal tick - axon is used in czech: á, é, í, ó, ú, ý)

I wish I could use the english layout actually, but I am too old to relearn z vs. y :-D

**Comment by Daeraxa** [comment]
I just mean that the double backticks in English are caused by the option to auto pair them, if you turn it off you only get one. My expectation is that when set to German pressing <kbd>shift</kbd> + <kbd>`</kbd> twice (or once followed by <kbd>space</kbd>) would also auto-pair - which it does on Linux. Is this not the case?

**Comment by JakubGIT** [comment]
> I just mean that the double backticks in English are caused by the option to auto pair them, if you turn it off you only get one. My expectation is that when set to German pressing shift + ` twice (or once followed by space) would also auto-pair - which it does on Linux. Is this not the case?

Yes, it is like that.

but an insteresting thing:

when 
- [x] `Auto-pair braces, parenthesis, quotations, etc.`
then the cursor is moved to the very beginning in the case described in this issue or after three backticks and the first key afterwards

when
- [ ] `Auto-pair braces, parenthesis, quotations, etc.`
then the cursor is moved to the very beginning immediately after the second backtick comes

**Comment by JackGruber** [comment]
In the meantime I noticed this does not happen in my Joplin which I use. But when I start it as dev version it does (same plugins installed).
The only difference is the note and tag number. Maybe some timing problem, because the dev version reacts faster? 
I can't see any other difference right now.

**Comment by shbach** [comment]
> Maybe some timing problem, because the dev version reacts faster?

Not quite the same, but in a [similar cursor jumping bug](https://github.com/laurent22/joplin/issues/5969#issuecomment-1078510949) I was sort of able to fix it by changing the timing of when I inserted/deleted text. Maybe all of these cursor jump bugs have something to do with timing.

**Comment by fabiorzfreitas** [comment]
As I described on #6460 (a duplicate), this bug seems to occur with all diacritics, i. e. ``` ` ```, `´`, `~` and `^`.

**Comment by Rolograaf** [comment]
Just discovered this also happens with shift-clicking the mouse in the editor. It jumps straight to the top of the buffer.

**Comment by RNBermudez** [comment]
I am having the same issue. 

Joplin version: 2.9.17 prod,win32
Platform: Windows
OS specifics: Windows 10, English Language
OS Keyboard: United States-International (this means [dead keys](https://en.wikipedia.org/wiki/Dead_key) are active)!
Physical Keyboard: US layout

This is happening with the following characters: `` ` ``, `'`, `~`, and `^`.

**Comment by laurent22** [comment]
> 

On which OS?

**Comment by RNBermudez** [comment]
> Platform: Windows
> OS specifics: Windows 10, English Language

Windows 10 Pro x64, Version 10.0.19044 Build 19044, specifically.

**Comment by fabiorzfreitas** [comment]
Not to make any demands on FOSS project, but this bug still happens to me. I'm sorry to say that I started looking for alternatives to Joplin, even though I love it.

Here's my info:

Joplin 2.9.17 (prod, win32)

Client ID: 0ee0b39f54d64204a5307a553a39aba2
Sync Version: 3
Profile Version: 41
Keychain Supported: Yes

Revision: a84a8e7

Windows 10 22H2 19045.2728

Keyboard: Brazilian Portuguese (ABNT-2)

**Comment by fabiorzfreitas** [comment]
I happened to be browsing my GitHub notifications and tried to see if the issue is still present.

It's not, now it works as expected, so this can be closed as Solved.
